api/admin/password_admin_authentication_provider.py

```diff
@@ -50,3 +83,37 @@ def sign_in(self, _db, request={}):
     def active_credentials(self, admin):
         # Admins who have a password are always active.
         return True
+
+    def generate_reset_password_token(self, admin: Admin, _db: Session) -> str:
+        secret_key = ConfigurationSetting.sitewide_secret(_db, Configuration.SECRET_KEY)
+
+        reset_password_token = admin.generate_reset_password_token(secret_key)
+
+        return reset_password_token
+
+    def send_reset_password_email(self, admin: Admin, reset_password_url: str) -> None:
+        subject = f"{AdminClientConfig.APP_NAME} - Reset password email"
+        sender = f"{AdminClientConfig.APP_NAME} <do-not-reply@thepalaceproject.org>"
```

Comment:
I think this email address is going to need to be configurable, so that we can make sure whatever SMTP server we are using can deliver emails from this address.